features/step_definitions/cli_steps.ts

```diff
@@ -1,94 +1,105 @@
-/* eslint-disable babel/new-cap */
-
 import { When, Then } from '../../'
 import { expect } from 'chai'
 import { normalizeText } from '../support/helpers'
 import stringArgv from 'string-argv'
 import Mustache from 'mustache'
+import {
+  valueOrDefault,
+  doesNotHaveValue,
+  doesHaveValue,
+} from '../../src/value_checker'
+import { World } from '../support/world'
 
-When(/^I run cucumber-js(?: with `(|.+)`)?$/, { timeout: 10000 }, function(
-  args
-) {
-  args = Mustache.render(args || '', this)
-  args = stringArgv(args || '')
-  return this.run(this.localExecutablePath, args)
-})
+When(
+  /^I run cucumber-js(?: with `(|.+)`)?$/,
+  { timeout: 10000 },
+  async function(this: World, args: string) {
+    const renderedArgs = Mustache.render(valueOrDefault(args, ''), this)
+    const stringArgs = stringArgv(renderedArgs)
+    return this.run(this.localExecutablePath, stringArgs)
+  }
+)
 
 When(
   /^I run cucumber-js with all formatters(?: and `(|.+)`)?$/,
   { timeout: 10000 },
-  function(args) {
-    if (!args) {
+  async function(this: World, args: string) {
+    if (doesNotHaveValue(args)) {
       args = ''
     }
     // message is always outputted as part of run
     const formats = ['json:json.out']
     args += ' ' + formats.map(f => `--format ${f}`).join(' ')
-    args = Mustache.render(args, this)
-    args = stringArgv(args)
-    return this.run(this.localExecutablePath, args)
+    const renderedArgs = Mustache.render(args, this)
+    const stringArgs = stringArgv(renderedArgs)
+    return this.run(this.localExecutablePath, stringArgs)
   }
 )
 
 When(
   /^I run cucumber-js \(installed (locally|globally)\)$/,
   { timeout: 10000 },
-  function(location) {
+  async function(this: World, location) {
```

Comment:
Is location always a string?

Reply:
Yep, going to create some tasks after initial merge to ensure we type everything